### DISCUSSION TITLE
schedule:list

### DISCUSSION THREAD

**Discussion by hooman-mirghasemi** [laravel/framework#42380] [Q&A] [answered]
sail art schedule:list

   TypeError 

  ReflectionFunction::__construct(): Argument #1 ($function) must be of type Closure|string, Modules\Advertisement\Schedule\ExpireOutdatedAds given

  at vendor/laravel/framework/src/Illuminate/Console/Scheduling/ScheduleListCommand.php:178
    174▕     private function getClosureLocation(CallbackEvent $event)
    175▕     {
    176▕         $function = new ReflectionFunction(tap((new ReflectionClass($event))->getProperty('callback'))
    177▕                         ->setAccessible(true)
  ➜ 178▕                         ->getValue($event));
    179▕ 
    180▕         return sprintf(
    181▕             '%s:%s',
    182▕             str_replace($this->laravel->basePath().DIRECTORY_SEPARATOR, '', $function->getFileName() ?: ''),

      +2 vendor frames 
  3   [internal]:0
      Illuminate\Console\Scheduling\ScheduleListCommand::Illuminate\Console\Scheduling\{closure}()

      +15 vendor frames 
  19  artisan:37
      Illuminate\Foundation\Console\Kernel::handle()

I got this error when i create invoke class according to document and my schadule work currectly. 
the problem only is with schedule:list

**Comment by korkoshko** [comment, marked as answer]
Hmm, the workaround is to set the class name as the description, for example:

```
$schedule
    ->call(new SomeScheduleTask())
    ->description(SomeScheduleTask::class)
    ->hourly();
```

See [#L88](https://github.com/laravel/framework/blob/9.x/src/Illuminate/Console/Scheduling/ScheduleListCommand.php#L88).

A simple framework solution this problem is to autofill the description in the `CallbackEvent ` constructor if it's an object, but i dont like that : )

**Reply by SKToke** [reply]
I still facing the same error with the description set.

**Reply by korkoshko** [reply]
@SKToke I fixed the current issue in this PR [#42400](https://github.com/laravel/framework/pull/42400). Wait for the release.
